### PR TITLE
DX: Restore PHPDoc to type rules workflow step

### PR DESCRIPTION
**Pull request by jrmajor**

### CODE REVIEW

**Review by julienfalque — APPROVED** [review]

**Comment by julienfalque** [comment]
Thank you @jrmajor, nice catch!

File: .php-cs-fixer.php-lowest.php [hunk 1]
```diff
@@ -24,12 +24,13 @@
...
     'src/DocBlock/Annotation.php',
+    'src/Doctrine/Annotation/Tokens.php',
     'src/Tokenizer/Tokens.php',
 ]);
 
 $config->setRules([
-    'phpdoc_to_param_type' => true, // EXPERIMENTAL rule, helping to ensure usage of 7.0+ typing
-    'phpdoc_to_return_type' => true, // EXPERIMENTAL rule, helping to ensure usage of 7.0+ typing
+    'phpdoc_to_param_type' => true,
+    'phpdoc_to_return_type' => true,
```
Comment (SpacePossum) [inline .php-cs-fixer.php-lowest.php:33]: why were these comments  removed, these seems valid to me?

File: src/Runner/Runner.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ final class Runner
      * @param list<FixerInterface>       $fixers
      */
     public function __construct(
-        $finder,
+        \Traversable $finder,
```
Comment (keradus) [inline src/Runner/Runner.php:71]: bc breaker
https://symfony.com/doc/current/contributing/code/bc.html#changing-classes

Comment (keradus) [reply src/Runner/Runner.php:71]: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6634

File: .php-cs-fixer.php-lowest.php (hunk 1 above)
Comment (jrmajor) [reply .php-cs-fixer.php-lowest.php:33]: I don't see the point of having them there, IMO that's information that should be in the rule description, not in the comments placed where the rule is used.

File: src/Runner/Runner.php (hunk 2 above)
Comment (jrmajor) [reply src/Runner/Runner.php:71]: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6634#issuecomment-1246044404

**Pull request merged**